collect_app/src/test/java/org/odk/collect/android/javarosawrapper/FakeFormController.java

```diff
@@ -0,0 +1,197 @@
+package org.odk.collect.android.javarosawrapper;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.form.api.FormEntryController;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.exception.JavaRosaException;
+import org.odk.collect.android.formentry.audit.AuditEventLogger;
+import org.odk.collect.android.utilities.StubFormController;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class FakeFormController extends StubFormController {
```

Comment:
Kotlin? :smile:

Reply:
I gave it a quick go, but there's a bunch of nullable code that needs to be revised. Let's hold that for now given the scale of other changes here?